### PR TITLE
[HOPSWORKS-745]: make warehouse readable by SparkSQL jobs

### PR DESCRIPTION
**Pull request by Limmen**
SparkSQL cannot impersonate the Hive super-user for reading the warehouse root directory. SparkSQL runs as the project-user and therefore the warehouse must be readable by project users for SparkSQL on Hive to work.

Tests: https://github.com/logicalclocks/hops-testing/pull/223

### CODE REVIEW

File: recipes/_configure.rb [hunk 1]
```diff
@@ -74,23 +74,33 @@
...
+end
+
 # Create hive scratchdir on hdfs
 hops_hdfs_directory node['hive2']['scratch_dir'] do
     action :create_as_superuser
     owner node['hive2']['user']
     group node['hive2']['group']
-    mode "1777"
+    mode "0777" #scratchdir must be read/write/executable by everyone for SparkSQL user-jobs to write there
     not_if ". #{node['hops']['home']}/sbin/set-env.sh && #{node['hops']['home']}/bin/hdfs dfs -test -d #{node['hive2']['scratch_dir']}"
```
Comment (SirOibaf) [inline recipes/_configure.rb]: Why did you remove the sticky bit? This makes that a user can delete someone else's dir, potentially killing a running query.

Comment (Limmen) [reply recipes/_configure.rb]: I don't know, I'll add it again

Comment (Limmen) [reply recipes/_configure.rb]: Done
